xdsl/tools/xdsl_tblgen.py

```diff
@@ -339,8 +339,7 @@ def _resolve_prop_constraint(self, rec: TblgenRecord | str) -> str:
         if "AnyAttrOf" in rec.superclasses:
             return textwrap.dedent(f"""
             AnyOf(
-                {",".join(self._resolve_prop_constraint(x["def"]) for x in rec["allowedAttributes"])}
-                )
+                ({",".join(self._resolve_prop_constraint(x["def"]) for x in rec["allowedAttributes"])})
```

Comment:
Will this work if there is only 1 entry? (I realise that this is a degenerate case and the corresponding `AnyOf` might also be broken

Reply:
I think that should also work, will try with a testcase

Reply:
I would expect that it would with a list and not with a tuple, but it makes more sense to me to do the len check and skip the AnyOf if there's only one element anyway.

Reply:
for the AnyAttrOf with 1 entry, I'm not sure if adding a hotfix in the conversion to json that removes the AnyOf is not confusing. Probably better to fix the TableGen file instead